File: java/samples/getSecurityLatestDividendRecord.java
import com.intrinio.invoker.ApiClient;
import com.intrinio.invoker.ApiException;
import com.intrinio.invoker.Configuration;
import com.intrinio.invoker.auth.*;
import com.intrinio.api.SecurityApi;
import com.intrinio.models.DividendRecord;

public class App
{
    public static void main( String[] args )
    {
        ApiClient defaultClient = Configuration.getDefaultApiClient();
        ApiKeyAuth auth = (ApiKeyAuth) defaultClient.getAuthentication("ApiKeyAuth");
        auth.setApiKey("YOUR API KEY");

        SecurityApi securityApi = new SecurityApi();

        String identifier = "AAPL"; // String | A Security identifier (Ticker, FIGI, ISIN, CUSIP, Intrinio ID)

        try {
            DividendRecord result = securityApi.getSecurityLatestDividendRecord(identifier);
            System.out.println(result);
        } catch (ApiException e) {
            System.err.println("Exception when calling SecurityApi#getSecurityLatestDividendRecord");
            e.printStackTrace();
        }
    }
}

File: ruby/samples/get_security_latest_dividend_record.rb
# Load the gem
require 'intrinio-sdk'
 
# Setup authorization
Intrinio.configure do |config|
  config.api_key['api_key'] = 'YOUR_API_KEY'
end
 
security_api = Intrinio::SecurityApi.new
 
identifier = "AAPL" # String | A Security identifier (Ticker, FIGI, ISIN, CUSIP, Intrinio ID)
 
begin
  result = security_api.get_security_latest_dividend_record(identifier)
  security = result.security

  puts "Security Summary"
  puts "-------------------------------------------------"
  puts "Name:       #{security.name}"
  puts "Code:       #{security.code}"
  puts "Company ID: #{security.company_id}"
  puts "Ticker:     #{security.ticker}"
  puts ""
  puts ""
  puts "Latest Dividend Record"
  puts "-------------------------------------------------"
  puts "Announcement date: #{result.announcement_date}"
  puts "Record date:       #{result.record_date}"
  puts "Pay date:          #{result.pay_date}"
  puts "Frequency:         #{result.frequency}"
  puts "Status:            #{result.status}"
  puts "Forward yield:     #{result.forward_yield}"
  puts "Forward rate:      #{result.forward_rate}"

rescue Intrinio::ApiError => e
  puts "Exception when calling SecurityApi->get_security_latest_dividend_record: #{e}"
end
